lib/services/role.go

```diff
@@ -118,41 +132,330 @@ type Access interface {
 type Role interface {
-	// Resource provides common resource methods
+	// Resource provides common resource methods.
 	Resource
-	// GetMaxSessionTTL is a maximum SSH or Web session TTL
-	GetMaxSessionTTL() Duration
-	// SetLogins sets logins for role
-	SetLogins(logins []string)
-	// GetLogins returns a list of linux logins allowed for this role
-	GetLogins() []string
-	// GetNodeLabels returns a list of matching nodes this role has access to
-	GetNodeLabels() map[string]string
-	// GetNamespaces returns a list of namespaces this role has access to
-	GetNamespaces() []string
-	// GetResources returns access to resources
-	GetResources() map[string][]string
-	// SetResource sets resource rule
-	SetResource(kind string, actions []string)
-	// RemoveResource deletes resource entry
-	RemoveResource(kind string)
-	// SetNodeLabels sets node labels for this rule
-	SetNodeLabels(labels map[string]string)
-	// SetMaxSessionTTL sets a maximum TTL for SSH or Web session
-	SetMaxSessionTTL(duration time.Duration)
-	// SetNamespaces sets a list of namespaces this role has access to
-	SetNamespaces(namespaces []string)
-	// CanForwardAgent returns true if this role is allowed
-	// to request agent forwarding
-	CanForwardAgent() bool
-	// SetForwardAgent sets forward agent property
-	SetForwardAgent(forwardAgent bool)
-	// CheckAndSetDefaults checks and set default values for missing fields.
+	// CheckAndSetDefaults checks and set default values for any missing fields.
 	CheckAndSetDefaults() error
-	// Equals returns true if roles are equal
+	// Equals returns true if the roles are equal.
 	Equals(other Role) bool
+
+	// GetMaxSessionTTL gets the maximum duration for a SSH or Web session.
+	GetMaxSessionTTL() Duration
+	// SetMaxSessionTTL sets the maximum duration for a SSH or Web session.
+	SetMaxSessionTTL(duration time.Duration)
+
+	// GetOption gets an OpenSSH boolean option.
+	GetOption(string) bool
+	// SetOption sets an OpenSSH boolean option.
+	SetOption(string, bool)
+
+	// GetLogins gets *nix system logins for allow or deny condition.
+	GetLogins(RoleConditionType) []string
+	// SetLogins sets *nix system logins for allow or deny condition.
+	SetLogins(RoleConditionType, []string)
+
+	// GetNamespaces gets a list of namespaces this role is allowed or denied access to.
+	GetNamespaces(RoleConditionType) []string
+	// GetNamespaces sets a list of namespaces this role is allowed or denied access to.
+	SetNamespaces(RoleConditionType, []string)
+
+	// GetNodeLabels gets the map of node labels this role is allowed or denied access to.
+	GetNodeLabels(RoleConditionType) map[string]string
+	// SetNodeLabels sets the map of node labels this role is allowed or denied access to.
+	SetNodeLabels(RoleConditionType, map[string]string)
+
+	// GetRules gets all allow or deny rules.
+	GetRules(rct RoleConditionType) []RoleRule
+	// SetRules sets an allow or deny rule.
+	SetRule(rct RoleConditionType, resources []string, verbs []string)
```

Comment:
I would argue that this interface is inconvenient (how do you delete a rule?) Maybe `GetRules() / SetRules()` is all we need, so you can update/delete multiple all rules at once?